### ISSUE TITLE
Future of food preservation.

### ISSUE THREAD

**Issue by nexusmrsep** [CleverRaven/Cataclysm-DDA#24145]
This is a discussion thread for **further development of food, food preserving methods and all related things**.
I'm making this tread, becouse of recently merged changes, to diagnose if and where there is room for more, and what is the direction it should go. There were some fine ideas scattered among merged PR's some of which were adopted on the go, some need to be done separately or go under discussion.

_Sum-up of recent changes:_
1. **immortal food overhaul** [PR #23986] - a big chunk of non-perishables became perishable, some with long periods of shelf life.
2. **better food decay indication/feedback** [PR #24057] - now, when you [e]xamine food, you will not only get it's shelf life time (default), but you will be informed how old it is. This however will require some skill in cooking or survival.
3. **underground termperature overhaul** [PR #24073]- basements, mines, caves as long as they are underground, have significantly lower temperatures then that above ground, and it is not weather dependant
4. **root cellars** [PR #24083] - a new construction you can make in your backyard, that works like an underground storage for food, it keeps it in a constant low temperature, and protected from weather conditions - people used to have those when there were no refrigerators.

What I want to achieve is a TODO list of things, that i will sum-up while this discussion will go ahead, be editing this post.

**Try to propose:**
- mechanics that would improve the game, it's depth, balance or playability in food-department
- recipes for food based on existing ingredients, that would achieve something (for example im planning a suggested PR on pickled eggs, that will add a new way to preserve them)

**Try NOT to propose:**
- reverts on introduced mechanics, spoil times unless with a proposition of better mechanic or solid data backing
- recipes for food harvested from the cookbook, just for the sake of it - there is already a ton of recipes and in my opinion only those serving a purpose (like preserving ingredients that were not preserved before in any way) might be worth considering.

Feel free to restate again your previous ideas in that matter here.

### **Ideas proposed:**

_Mechanics:_
1. Food stored outside **proper furniture** would get penalty (extra decay), to discourage "floor-pasta" and encourage proper storage. (where to go from that? - should it be a special 'pantry' furniture or any 'kitchen' furniture would be fine?)
- **food flags** (maybe: frozen, cool, dry, sealed) and then checks for proper storages based on flags set.

2. **Packaging**. Maybe packing food should have some impact? There is no way to use recykled bags and other pacing material, but that might be an idea to talk about at least. How would it work and why?
- packaging may prevent vermin/carrion effects

3. **Food irradiator** - industrial machinery that sterilizes perishable food converting it into non-perishable (or perishable with longer shelf-life) irradiated food (that already exists in-game). High energy consumption might require a separate generator/reactor (plutonium?). Components/elements: generator/reactor, food tray, control panel, rad-emmiter, rad-scrubber (cleans up secondary radiation) . Possible side-effects, radiation based. Game balance control: energy usage, irradiation factor --> scrubber energy/material consumption.

4. [ PR #24209 **MERGED** ] Introduce **RNG to decay state** of food spawned in game **originating from before cataclysm** to simulate different storage conditions. Some food might be less some more decayed when found. (20% +/- change?)

5. [ PR #24480  ]**Field dressing** of hunted game as a method preventing fast meat spoilage, plus corpse's rot should affect meat's rot if it was not field dressed before

6. [ PR #24228 ] **Freezer** - a variant of mini-fridge that is designed to keep food under zero degrees Celsius. 
- **FROZEN** and **NOFREEZE** flags as potential method to simulate thawing process and food that doesn't like to be frozen

7. possibility to **eat food past rotten state** and old food with increasing risks, also: degradation of nutrition values and fun of old food

8. embedding recipe components (ex. carrots) to the resulting product (ex. pickled vegetables (carrots)) for more specific use in further recipes.

8. possibility of jarred/canned food to spoil, but not linear way, just a rng chance rate (that maybe starts after content's nominal spoil time ends)

9. Revisit #23141

10. #24307 - **Smoking rack interactions expansion**, ability to batch smoke/dehydrate food while away form the smoking rack

11. Add food composition system: **CARBS/FAT/PROTEIN** and appropriate needs and effects for each.

12. Migrate comestibles nutrient values to kcal values, made possible by #14613. Display nutrient values in kcal in game menus.

_Recipes:_

1. [edit: PR #24149 MERGED] ~~Pickled eggs - in addition to pickling cucumbers/pickes, veggies and meat, to preserve them~~
2 [#24225] Fix for garlic clove not rotting as it should.

**Comment by TechyBen** [comment]
I'd recommend not making kitchen furniture a special case. But allowing any non drafty/moldy cupboard to count for storage. Kitchen furniture would still be artistic etc, but functionally it is the same as any other place with doors + storage. Where as wire baskets would not have the same effect on food.

Indicating this to a player would help. Such as "you put your apple in the *clean* cupboard" or "you drop your salami in the *cool* cellar".

Would this allow some rotten food in jars/bags/cans to go rotten but not generate pests? As it would be rotten, but still sealed? Whereas unsealed rotten food would? It would add a layer of storage vs protection vs checking. You would need to check for quality of food, but checking for pests is a different dynamic, and the two systems have their own needs and requirements, instead of both being exactly the same. :)

**Comment by TechyBen** [comment]
That seems a good idea and reasonable, that the game is assumed to do the micromanaging. We don't need to simulate drying out or cleaning an area IMO. Beyond the current game items (rote/splatter etc). Adding a "moist" tile would complicate things, along with adding the need for coal or dehumidifying items.

It can easily be assumed any non-damaged surface/container is "clean" and "dry". The floor-pasta seems more a gameplay mechanic choice and one of simplicity. While storing food on the floor is possible (see grain silos), in real life is rarely done. In addition, a player is more likely to think "floor bad, table good" for storing food, and reflecting that in the code (either requiring cupboards, or relaxing the rules to allow tables) will IMO better reflect the expectations of a player.

**Comment by RadHazard** [comment]
I agree with the other points.  Before we start nerfing rot timers to account for average apocalypse conditions, we need to make sure the player has tools to keep their own food in better-than-average conditions.

My suggestion would be to allow containers like boxes, bags, jars, etc. to increase the shelf-life of food and prevent/reduce pest spawns (probably depending on the container - jars and cans are virtually pest-proof but boxes and bags aren't).  Realistically this should stack with the "pantry" idea, but I'm not sure if that ends up being too much micromanagement for the player.

To support that, we will also need the ability to put food back into the various containers without having a recipe for every single combination.  Vacuum-sealing, canning, and jarring should also be expanded to a system that can be used for most kind of foods (maybe with flags to control which items can be preserved with those methods).

There's also a need for player-craftable food containers for people who can't find real ones (e.g. wilderness runs).  They don't need to be able to match pre-cataclysm containers in effectiveness, but they should be easily crafted (things like wooden boxes).

**Comment by kevingranade** [comment]
>  allowing any non drafty/moldy cupboard to count for storage.

This misses the point, cupboards do not protect food from temperature, humidity, or small vermin (in theory they protect from large vermin such as rats).

> Why wouldn't a survivor be capable of storing his food in decent conditions?

No one has said they're not capable, but that they need to take measures to make it happen.
> We shouldnt assume that every nook and crany is automaticly infested with insects, maggots and mice.

We absolutely should, because in reality, in the area in question, they are.  Modern homes have a lot of features present to keep out vermin, but if unmaintained, those measures are going to be quickly defeated.

>  Are we really going to force the player to think about packaging their food? Is extreme realism really the way to go?

Yes and yes.  If you want to maintain a store of permafood, you need to put some thought into it. Likewise were putting a fairly hard cap on the lifespan of pre-cataclysm food that isn't stored properly to push the player to acquire it elsewhere as they transition from short-term scavenging to long term sustainability.  Or they can keep scavenging preserved foods nearly indefinitely, there are a lot of options.

> Or perhaps a simple approach where leaving some canvas bags or plastic containers on the floor ensures that the player automaticly puts them to use when storing food.

How is this semi-magical interaction any more streamlined than building a larder to store your food in?

> To keep things more specific: ...

Please provide sources for these mechanisms being useful.  Also I don't even know what you mean about "using salt", are you just scattering handfuls of salt around your food storage area?  I'm very sceptical that's going to meaningfully impact humidity levels.

> lastly I think storage in cars deserves some serious attention.

Give it some attention then, propose some craftable food storage areas 5hat can go in a vehicle.  Or don't, it's perfectly fine if nomads have more trouble maintaining their food supply.

**Comment by DemAvalon** [comment]
> Give it some attention then, propose some craftable food storage areas 5hat can go in a vehicle. Or don't, it's perfectly fine if nomads have more trouble maintaining their food supply.

How about a fridge?, like those we already have in every single house?, I remember someone mentioning in the forums that there was a mod that allowed the player to unplug fridges (on the construction menu) and that turned them into an item like the mini fridge (but with much more room)

Also mobile kitchen units, FOODCO kitchen buddy and on-board chemistry labs should count as proper storage places for food (the same as a pantry)

**Comment by GroeneAppel** [comment]
>No one has said they're not capable, but that they need to take measures to make it happen.

That is fair, but this means that some form of extra action has to be taken to reduce this. 

>We absolutely should, because in reality, in the area in question, they are. Modern homes have a lot of features present to keep out vermin, but if unmaintained, those measures are going to be quickly defeated.

Here I have doubts. I see little reason that every single area of a building or car is automaticly going to be infested with vermin or other living organisms that will spoil food quickly. Any chance that you have some sources for that?
Regardless, would players be required to 'prepare' an area beforehand by cleaning it? Aside from a larder or the use of coal that is. Kitchen cabinets, or storage areas in cars may already exist beforehand. If we follow the thought that all areas are infested, these areas will need cleaning. This would add some more use to soap and detergent. This is turn would turn an area 'clean' and better for storage. 

>How is this semi-magical interaction any more streamlined than building a larder to store your food in?

Quite simple. Building a larder requires construction, tools and materials. The use of for example plastic containers is a great _alternative_ to this. Conserving food should not be linked to one single method. Even more so, the use of both a larder and proper containers, should be rewarded with less spoilage. After all, vermin will have a somewhat harder time to reach something in a plastic container/larder

>Please provide sources for these mechanisms being useful. Also I don't even know what you mean about "using salt", are you just scattering handfuls of salt around your food storage area? I'm very sceptical that's going to meaningfully impact humidity levels.

This one is harder. Coal and salt have long been used to help reduce humidity and ensure longer lasting food. Amusingly I keep a bag of coals in my fridge right at this moment, as it helps reduce the moisture in there. What kind of sources are you looking for? Research papers are typically much more specific (and not accesible for many due to the pay-to-read structure). However pulling some recommendation on food storage from google feels much less reliable. 

Regardless:
http://www.andatech.com.au/resource/reduce-indoor-humidity-without-dehumidifier/
https://www.saimm.co.za/Journal/v102n07p417.pdf
https://www.omicsonline.org/open-access/effect-of-wood-charcoal-powder-on-rate-of-microbial-production-of-lactic-acid-in-dehulled-and-undehulled-vigna-unguiculata-pastes-2155-9600-1000587.php?aid=87336

Overall these articles show that coal strongly helps reduce moisture and bacteria production in an area. In addition one can find many sources on the internet with a simple search that show the use of coal (or salt) to reduce moisture in an area. 

>Give it some attention then, propose some craftable food storage areas 5hat can go in a vehicle. Or don't, it's perfectly fine if nomads have more trouble maintaining their food supply.

The existing fridge is an absolutely fine option here. Alternatively a wooden cupboard (12x 2x4?) OR metal cupboard (steel frame) could form good alternatives to storing food in a car. The interresting subject to food storage in cars is that of heat and cold due to the environment. The concept of cars boiling those inside them on a sunny day is not a strange one. However when those cars are running, their airconditioning can keep the inside temperature and humiditiy more than enjoyable. This can and should be utilized in Cataclysm. A simple option to turn on airconditioning would be a good addition here. 

Currently entering a car in deep winter will offer the player a comfortable temperature (provided the car has no holes in it). However this should not be the case without some proper heating or airconditioning. Adding this option would, much like the car stereo, offer players a choice in reducing the spoilage of their food and perhaps more importantly; their own comfort. 

Anyway, overall i'm happy that we can enter a discussion on this topic. Even though the system in cataclysm will never be perfect. It's pleasant to know that we can atleast discuss it. Food spoilage is an interesting, if not complex topic, but one well worth discussing.

**Comment by TechyBen** [comment]
>This misses the point, cupboards do not protect food from temperature, humidity, or small vermin (in theory they protect from large vermin such as rats).
Yep. I was only commenting on when/what we choose as a place to store should automatically be considered clean and dry. If we only consider functioning fridges and the proposed player constructed larder, then that's great.

I'm merely suggesting that adding damp/dirty tile flags and player interaction is possible (other games such as Rimworld have this and work well!) but can be rather hard for a player to manage. But if it is being implemented, then with the NPC camp PR, would be manageable, as it would be offloading the work to NPCS. But if as a single player, cleaning/drying constantly in game could get tedious. Thanks!

**Comment by Tharn** [comment]
I can only repeat myself. Tupperware and other sealable plastic boxes are **ubiquitous** today. No reason they shouldn't be in the future. I'm sorry if this is too easy or too obvious of a solution, but if we're going for realism, let's be fair about it.

Vermin are definitely on the rise during the cataclysm, but if a house is still intact and no windows are broken, for example, there's no reason to assume it'll be any more prone to infestation than a house is today. Well, depending on construction. Plants and vines may crack walls or bad plumbing may wear.

I do think cupboards etc. are not enough to protect from all vermin. Moths for example have the uncanny ability to get in almost anywhere that's not properly sealed. They can and will ruin rice, flour, sugar and grains.

I think in this context, it may be a good idea to have several storage modifiers, like `frozen`, `cool`, `dry`, `air-tight`. Not all foods should require all four to preserve, but most will require at least two. You may use a combination of storage to attain these modifiers. A homemade cellar will probably be cool but humid, requiring use of good storage containers. Fridges also get humid as soon as you store unsealed perishables like fruits, veg and meat in there.

I think the next big thing could be that these cardboard boxes and plastic bags that most food comes in may serve the function of extending shelf life. So if you have a habit of (U)nloading all your food into piles, you save space but spoilage will accelerate. And that sealable plastic boxes could be a thing.

**Comment by Kelenius** [comment]
I have mixed thoughts about the latest change. I'm going to try to break it down.

First off, I do not like how quickly the change came - one of the points of #23986 was that food is now considered to be always stored in bad conditions, without allowing the player to actually create good conditions. The old situation wasn't exactly realistic, because food in the houses isn't going to last forever, but it's also not realistic now, because realistically you should be able to put your rice into a sealed container and it will stay like that for a very long time. Essentially, we went from "food is always considered to be stored in good conditions" to "food is always considered to be stored in bad conditions". I think that the ability to protect food from the elements should have been added before lowering the rot timers.

Secondly, I believe that any food that can be stored for more than an year shouldn't rot at all. From the gameplay perspective, if the players allows food in their storage to rot away after keeping it for so long, food is probably not an issue for them. It just clutters the menus. From the realism perspective, this idea also has its flaws - for example, all rice in the game takes exactly the same time to start rotting; for the food with short rot timers it's not very noticeable, but food that can be stored for years should have a very large variance in this timer - taking rice for example, if it's dry and contained, it will last for years, but in bad conditions it can become bad a lot faster.

I think that a better idea for gradually removing the food from scavenging would be to introduce a scaling chance that some types of food generated on the map wouldn't spawn or would spawn spoiled - e.g. the chance that starts at 0% after an year and goes to 100% after three years, so after an year you start randomly finding spoiled rice, and after three years all rice that you can find is spoiled, but the rice that you already have with you won't go bad - we assume that the player character is keeping it dry.

**Comment by Zireael07** [comment]
> I have mixed thoughts about the latest change

What latest change are you referring to?

**Comment by Kelenius** [comment]
>What latest change are you referring to?

#23986

**Comment by DemAvalon** [comment]
Actually, we have irradiated food, right? and we already have atomic coffee machines right?, so why not make a "Portable" food irradiator, a machine that you weld to a vehicle and drains loads of power when active and it just sterilize all food placed within it (after like two hours or so) increasing their shelf life dramaticly.

It could be a rare spawn in groceries or something (factories?) or possible craftable with high mechanics and electronics, using the plutonium batteries or the microreactor CBM as ingredients.

**Comment by nexusmrsep** [comment]
@DemAvalon That is a brilliant, fantastic idea, although I'd prefer it was a food factory industrial machinery, as citizens don't walk or drive around withe a portable hi-power x-ray machine. I already have few implementation ideas in my head, including possible radiation based side effects, etc. Definitly a queued PR on my side. Would need food factory json'ized map also. Ideas welcomed.

**Comment by Tharn** [comment]
Also, irradiated food doesn't last forever. It's meant to extend shelf life, but what does that actually mean? I've read reports of shelf life of irradiated shellfish being around 6-7 days, irradiated and vacuum-packed papaya being 60 days.

**Comment by nexusmrsep** [comment]
I updated the initial post.
@TechyBen great site, I almost feel like those pictures are from CDDA.
Also packaging food as a way to prevent newly introduced carrion effects is also great. Noting it down.

**Comment by nexusmrsep** [comment]
> Also, irradiated food doesn't last forever. It's meant to extend shelf life
So (as always) there are two ways to approach that:
1. go SCI-FI and allow irradiated food to be non-erishable (like now), but to counterbalance this limit the use of irradiator machinery the way i described in initial post above,
2. go RL, and make irradiated food perishable, but on different set of long(ish) shelf-life values, and loosen the grip on irradiator machinery, making it more common and easier to find & use (portable? vehicle part?)
I see benefits in both.

**Comment by TechyBen** [comment]
Yeah, I always though irradiated fruit and veg just gets a few extra days? And that the long shelf life in game was a missed update from when rotting/aging got introduced to the game? I cannot find any data on length of spoilage for irradiated foods, just "it helps", or "_up to double_", and mainly for meat. Already dried/preserved fruit and veg may get longer, but currently dried fruit/veg in game is already long term... However, when I've got irradiated *fresh* strawberries/salad (could also be UV treated in this country) it only adds a couple of days. So I guess we just need to decide which tech or process is being used. I'm no expert there.

@nexusmrsep Hahaha, yep, it's making me want to go there on holiday!
But I am also surprised that even up to July, not all the food on the shelves had been ransacked. It's very location dependant. Note the pig, the store it took up home in is totally empty, from the pig. The store (supposedly) locked, where only a single rat or so got in, has a couple of chocolate bars missing.

Certain foods could not be eaten in bulk by the animals large enough to get to them. Such as sugary sweets and rats. However, population explosions of rats/mice would destroy those stores quickly (see Mice Plague in Australian grain silos! :O ).

So it would depend on how we wish to simulate/artistically paint New England. Do we say closed grocery stores are rat proof but not cockroach proof? Or do we assume they are all run down and damaged from the first few hours of the fight?

**Comment by DemAvalon** [comment]
I like the more RL approach, so instead of being the ultimate solution its then a excellent tool to aid food preservation, so that when all is combined (proper packaging, proper food storage and irradiation) it makes for food that will last a real long time.

**Comment by nexusmrsep** [comment]
For RL approach I need some values, so I've done some research. There is ton of sources about food irradiating (see for yourselves, very illuminating material) , but most of them are useless and not giving any values about shelf life impact or comparisons. Found fome science articles, not many though...

**Few abstracts:**
`For example, according to the Council for Agricultural Science and Technology, irradiating strawberries extends their refrigerated shelf-life to up to three weeks without decay or shrinkage, versus three to five days for untreated berries.`

`Gamma irradiation was employed to restrain potato sprouting and kill pests in grain. Irradiation proved to be extremely beneficial in terms of prolonging the fruit and vegetable shelf life by 3–5 times.`

`The control bananas ripened within 6 days while the gamma irradiated bananas ripened within 26 days, indicating that the shelf life of banana was extended by 20 days thereby delaying banana ripening. `

`Mango in room temperature doubled its shelf life (on average). Average days taken to ripening 12-->22, average shelf life (90% batch overripe) 14-->25.`

`Dose range of 1.5–1.7kGy extended the storage life of pear by 14 days under ambient conditions. Control unirradiated pears were almost fully ripe within 8 days, while as the pears irradiated in the dose range of 1.5–1.7kGy were fully ripe within 22 days of ambient storage. Irradiation dose of 1.5–1.7kGy significantly inhibited the decaying of pears up to 16 days of ambient storage.` [ambient=room temp.]

I do not believe that i will find more, aspecialy for each and every items that had irradiated variant in game, so in my opinion it's safe to assume at least **_tripling spoil time for irradiated food_** will be fine. 

Objections? Thoughts?
P.S. Of course changing irradiated food spoil time must be in coordination with adding the machinery itself. This however still raises questions about portability as typical irradiation plant looks like one presented [HERE](http://www.barc.gov.in/bsg/ftd/faq2.html) or [HERE](https://www.iaea.org/newscenter/news/new-iaea-collaborating-centre-to-enhance-irradiation-research-safety-and-training-in-mexico) or [HERE](https://cna.ca/wp-content/uploads/2014/05/Nordion-irradiator.jpg). Rather not a household equipment however to look at it. There would be more then a pinch of SF needed to accept reducing it size to something portable and common. Which is a decision to make ... or not to make.

**Comment by Tharn** [comment]
Personally, I would err on the side of longer spoil times, like 4x. I would also not add this radiation machine. Don't really see the use for it, when 1) better modes of preservation exist, 2) this one should realistically involve plutonium cells and 3) we're shrinking down an industrial complex into a more or less hand-held device.

I would rather see more recipes for primitive modes of preservation, such as being able to dry more foods like fruit.

**Comment by DemAvalon** [comment]
Why not both?, you could have the factory one be safe, reliable and it has no adverse effects on the food, while you could have the "portable" (something you have to weld to a vehicle) be a piece of junk that uses loads of power, leaves radiation behind (like atomic coffee, for example) and could even mutate fruit (add a mutant fruit item that could have all kinds of effects when ingested) but the trade off is that you can put it on your death mobile, maybe even make it use plutonium slurry or other radioactive waste.

Also @Tharn you can already dehydrate fruit (and vegetables and meat) using a smoking rack(it counts as a charcoal smoker for recipes), that is pretty primitive.

**Comment by nexusmrsep** [comment]
@Tharn 

1. x4 would be acceptable too, in my opinion, based on those sources,
2. This irradiated food didn't come from thin air. Irradiating machine is a must in my opinion, only its form and size is to be discussed. Also not adding a rad-machine while changing rad-food from non-perishable to perishable would lead to spoiled rad-food with no way to replenish it. It might be justified by death of thechnology, but there is no sophisticated tech in using radiation.
3. Plutonium reqirement can be balanced in many ways - for example by simply adding a single plutonium rod for a lifetime of machine's need. Or convert it to energy, but deduct this energy at low-rate. RL process just exposes food to Cobalt-60 and basicly nothing else is required. Radiation protection required and safety standards is what makes this method more industrial. Else we'd have a Cobalt-60 home irradiator in a microwave format.
4. I'm not against shrinking myself, but as stated in nr 3. I'd rather think that believeable shrinking limit would result in a fridge-sized equipment with thick lead walls, mounted plutonium rod, with a mechanism to cut-off rad-rays, and a tray for a single product to be placed for irradiation.
5. Even if small equipment like in nr 4. would be implemented I'd still go with a food factory having either few of those or a larger version, just for RP reasons.
6. There is also a non-plutonium approach - electron beam bombardment. See [WIKI](https://en.wikipedia.org/wiki/Food_irradiation#Gamma_irradiation). "This system uses electrical energy and can be powered on and off." Hence, translating it into the game it might not use plutonium but electricity (vehicle part). Option to think about.

**Comment by DemAvalon** [comment]
> I'm not against shrinking myself, but as stated in nr 3. I'd rather think that believeable shrinking limit would result in a fridge-sized equipment with thick lead walls, mounted plutonium rod, with a mechanism to cut-off rad-rays, and a tray for a single product to be placed for irradiation.

>Even if small equipment like in nr 4. would be implemented I'd still go with a food factory having either few of those or a larger version, just for RP reasons.

I think that is a great way to go about it, have food factories normally have the normal huge one, but rarely they have the "compact" one, and the description could justify by saying its cutting edge technology, and the factories were getting some to test it out, but then the cataclysm happened, so they never really installed those, so you could have them in a room (something like the final room in labs) that has some kind of challenge even (like having to find the factory manager's ID card or something to get in) so it could feel very rewarding to explore the factories.

Although that is probably out of the scope here, sorry.

**Comment by plazma-rush** [comment]
Irradiation plants are large so that they can process large amounts of food at once. It would, in theory, be easy to scale it down.

Just spitballing ideas, but what if you could salvage an X-ray machine from a hospital, modify it to increase its power, attach it to a vehicle to use its power supply, and then use it to irradiate food? This isn't outside the realm of possibility, as X-rays are sometimes used to irradiate food and sterilize equipment.

**Comment by Amariithynar** [comment]
@nexusmrsep In both this thread and the original you linked back to (where Kevin said, "Please stick with this, just adjust effective spoilage times assuming that storage is never ideal. Better storage mechanisms can be added later." and was not countermanded at all, that you said was "flatly wrong" when @Kelenius noted it, btw) you seem to be incessantly pushing not for the actual *spoilage* time of foods, when they actually have rotted and are no longer edible, but instead more along the lines of the FDA's food safety recommendations, which replaces food because we as a society can afford to be picky about what we serve and consume in an effort to reduce food poisoning cases down to an effective zero as much as possible.

As a cook in the food industry, no, you do not need commercial level equipment to produce foods that are as long-lasting as those made on an industrial scale. The vast majority of the extended life comes from either the process- freeze drying, for example, just requiring constant airflow and desiccant to dry the air within the sealed space- or preservatives, which we have access to but don't on a personal scale because of the deleterious costs associated with their production. When you no longer care about ownership of items or ensuring you do not disturb neighbours, you can easily produce similar conditions to create what you need. Here's one enterprising individual who did just that using some basic physics knowledge, welding, and easily obtained materials; it'd have been even easier if he didn't have to pay for what he used, and could do the welding himself, like Survivors can. http://www.sciencemadness.org/talk/viewthread.php?tid=14652#pid190003

Foods do not go rotten immediately. In fact, regarding meat, even the so-called fresh cuts obtained in a market aren't fresh, but have already been aged for some time, often well over a month, before being put on sale. The process of dry aging is also incredibly simple, not something that requires an industrial process (which the Survivor can replicate themselves anyways, given enough scavenged materials and the determination to do so); Hang a cut of meat in a cool, dry room (which is all a refrigerator was originally meant to simulate), and let it age. After a month, it's reached the first step of aging. After two, it's reached prime age and is ready to be cut down from primals and sub-primals into the cuts we're all far more familiar with. You trim off the outside inch or two, and the meat within is ready. No seasonings needed, though butchers used to use salt blocks or charcoal (as has already been mentioned) to help remove moisture from the air... but even with those relatively primitive methods, meat kept for a very long time. Do you know what spoils meat the fastest? Leaving the guts in, so that the toxins carried inside of them can spread to the rest of the meat, as the barriers to the gut flora spreading are gone with the animal dead. Same with leaving the hide on and not letting the moisture escape. This is still done today in many parts of the world where there isn't such a system in place for easily accessible, pre-packaged foods; Scott Rea of the Scott Rea Project has a game hunter/butcher who he gets his venison and rabbits from, and you often see the carcasses hanging in his butchery just like that. 

As for vegetables and fruit? Yeah, store-bought fruit that was picked months ago and then stored in warehouses until being shipped to the store to be sold to you will only last a few days more on your counter or in your fridge. Ripe fruit, picked right off the tree? That typically lasts a bit longer, and is easily desiccated, which keeps basically indefinitely; stored in a dry cool place, dried fruit will last for months on end. True, apricots and other delicate fruits don't last very long even freshly picked, but they're also easily dehydrated, and again, once dehydrated, last for months at a time. Same if boiled down into a jam (breaking down cell walls, rendering out most of the water, adding lots of sugar as preservative, jarred and cooked at high temperature and pressure). 

And all this isn't even touching on the more general uses of chemical treating to preserve foods. Honey quite simply doesn't rot, period, because it's basically pure sugar; it has to be hydrated to rot. Fruit was commonly 'pickled' in honey for long term storage, especially by merchants traveling over Europe and Asia to sell their local produce elsewhere. Sugar Cane originally came to Europe over such trade routes, preserved like this. Lye was used to preserve fish in Scandinavia (the lye saponifies the fats and makes it generally very basic, which bacteria do not like- they prefer a moderately acidic environment). Salt has been used for millennia, both in liquid form as brine (used in pickling) and also as an additive to hang-drying methods to assist the desiccation process. Canning/pressure cooking for sterilization is only required if you aren't using a high acid ingredient (strawberries are, for example, notoriously easy, only requiring a short boil in their jar in an open pot, and no added preservatives) or aren't adding preservatives like vinegar or brine (which, for example, cucumbers need). 

And even then, going back to medieval times, large quantities of spices, or just strong spices, were used to mask the bad flavour of food that was rotting; it was still quite edible. Or the "Perpetual Stew" model of "preservation", where you have a stew made of basically anything and everything that never fully empties and constantly has new ingredients added to it. Some fraction of the ingredients are not used up but remain in the pot for months, even years, in some cases, much like how sourdoughs are made with sourdough starters that are traditionally torn-off pieces of someone else's sourdough starter, passed down through generations, all starting from a simple flour and water mixture carefully decanted and discarding the majority of the base and feeding it to get the original starter from airborne yeast, then every so often feeding it again, but otherwise just left to sit in a container in a (medieval standards) cool dry place until they made yeast bread.

Hell, even *broth* can be easily concentrated down and salted to a point it lasts a long time, and from then can even be spread on a tray and dried out for a day or two, and the remaining residue scraped off and ground down to a powder with a mortar and pestle for extremely long term storage.

Really dislike it when people treat foods like they're some sort of delicate thing that rots to inedibility if you just glance at it the wrong way, and that treat our digestive systems as if they're super delicate, not able to eat a wide variety of things with a wide variety of stages of deterioration, to survive. Anyways, rant over. 


Irradiating food is quite simple, needing any form of ionizing radiation; you could easily rig up a plutonium cell with some scrap, sheet metal, and a fair amount of lead to act as shielding casing, for example, or even just rig it up to a battery or other vehicular source of energy; this would be less efficient than a commercially available model, though, and would take more time to emit the same amount of energy in the same direction, as the more precisely machined and constructed commercial machine, so recipes using such a makeshift machine should take more time. It's also not something that should be a crafting recipe, per se, either, but should be treated more like a vat where you set it up to "ferment" into the final product, just instead of fermenting, it's irradiating.

**Comment by FulcrumA** [comment]
A few more things I've noticed regarding spoilage and preservation:

Many vegetables cannot be preserved in form other than generic vegetable matter - which is unsuitable for many recipes unlike normally preserved form of particular vegetables would be. Checking generic-making recipes like "pickled vegetable", "canned veggy" and "vacuum-packed veggy slice" may be useful here. While some recipes using particular vegetables also use the generic preserved kind, some vegs that can be "generalized" may or may not be usable directly in some of those recipes when they should/shouldn't (depending on the recipe).

Minifridge seems painfully inadequate. Normally I didn't really use it much, but now that it became much more important with far many things spoiling and fast at that I notice that it seems to only slightly lower the temperature compared to the ambient one outside rather than keeping it at low, cold temperature normally any fridge would (at greater or lower expenditure of power depending on surrounding). That or it just cools things badly in general as in summer meats stored in a fridge get rotten in a manner of a few days - while IRL meat certainly could lose a lot of its quality, but would be edible after thorough cooking assuming it wasn't already somewhat old once bought/acquired.
Assuming here that minifridge is of quality similar to full-sized home appliances (quality of minifridges varies but many commercial models are comparable, just, well, small).

Ability to deconstruct said full-sized fridges to drop an item rather than just parts certainly also should be considered (as could be said for many furnitures that could be disassembled with simple disassembling and require item butchering/disassembly to drop their components but that's a separate issue).
Maybe fridges could be even improved, and outright quality freezers found for the end-game where player can build/acquire reliable sources for constant flow of electricity. 

Garlic cloves do not spoil. Since they're not seeds with a hard shell, just pieces of a garlic bulb which does spoil, they should spoil as well.

**Comment by nexusmrsep** [comment]
@Amariithynar 
First of all, thanks for sharing your thoughts, thats great comment with solid amount of correct concepts, many of which I am familiar with, even from personal experience.

But:
> In both this thread and the original you linked back to (where Kevin said, "Please stick with this, just adjust effective spoilage times assuming that storage is never ideal. Better storage mechanisms can be added later." and was not countermanded at all, that you said was "flatly wrong" when @Kelenius noted it, btw) you seem to be incessantly pushing not for the actual spoilage time of foods, when they actually have rotted and are no longer edible, but instead more along the lines of the FDA's food safety recommendations, which replaces food because we as a society can afford to be picky about what we serve and consume in an effort to reduce food poisoning cases down to an effective zero as much as possible.

Nope. Flat wrong. Again...

You quote the first sentence of @kevingranade comment, but not the second one. Why? I almost think you did that on purpose to create false image of what truly was said. Let me remind you what he said there:
> As for what value should go in the json, that should be determined by what can most conveniently derived from irl sources, so if your sources say, "this will be good for one year", then put a number related to that one year figure in the json (your "rounding" scheme is fine, just don't do something like, "source says one year, but since durations are non-ideal that becomes one season").

Please don't create false image for other people, by putting things out of context. You are not making it any better by doing so.

If this was a misunderstanding however, let my put it in to a context.

In the begining we must understand how game mechanics treat food and what is what in it's core. I sincerely believe that we may not understand each other sometimes becouse we have different points of view as RL is being translated to in-game mechanics. You may see something from 'outside' while i try to look from the 'inside'. I and every contributor must account for actual possibility of conversion RL mechanics to in-game mechanics, and also have in mind that no simulation is perfect and there is a level of depth that cannot be crossed. Thats what 'simulating' is all about: how to 'degrade' RL to a mechanic that imitates it on an accepted level. Remember that we always balance between simplicity and depth. Trend is to go towards the depth, but not every bit is worth such translation and not every bit even can be represented by simulation either. You have to sometimes cut some slack on things.
I hope that what i say below clears those misunderstanding, as i will not repeat myself over and over again for the same reasons.

What is spoil time? Time in which food can be safely eaten.

1. True spoil time is variable. All things you mention affect it. Most common: processing, storage.
2. True spoil time must be calculated in-game to account for storage, processing.
3. You need base values to calculate true spoil time, so you need nominal spoil time.
3. True spoil time is not the nominal spoil time that game tells you about, but you must know nominal time to have a concept of what to expect. You, as a player, want to affect nominal value as much as possible when it becomes true spoil time (rot/decay) to keep food for longer time. This incentive creates fun and gameplay.
4. Nominal spoil time is an abstract spoil time that asumes ceteris paribus constant ambient conditions. Those conditions for game mechanics were defined as: constant room temperature.
5. Initial discussion mentioned storage conditions only becouse I was asking if they are (or not) part of ambient conditions for determining nominal spoil time. They were not, as the quote says.
6. I used sources. Multiple if possible. Excluded recomended "eat by date" etc. sources. Excluded sources with "refrigerator shelf-life". Excluded "stored in basement", "stored in humidity <2%", excluded EVERYTHING that stated non-nominal conditions. Excluded "opinion" sources as biased - I'm not working on opinions, I'm working on data (unless there is no data). Sources nevertheless. So I believe that they also don't assume you keep your rice in a damp backpack in the rain, but in normal household conditions - "ambient". I believe those sources give average data. Individual items IRL might not always represent average shelf-life. Put 10 apples in a row, from the same tree, and not every one will spoil at the same exact day. But on average - whole different story. 
7. When you have solid initial nominal rot - and I believe in most cases we now have - you calculate the rest.
8. Processing - in game - is achieved by recipes and their resulting items, they have their own spoil time based again on sources, and again this base value is used for all further calculations to account for other conditions.
9. Storage conditions - this is accounted for mostly by discrete mechanics, like temperature rot modificators - which btw is also a state-of-the-art achievment as it includes changing weather conditions and RL temperature influence on food. I strongly encourage everyone intrested to read the code directly on that topic.
Secondly it is achieved by recipes too and the use of certain tools: you already have your vacuum sealer for dry freezing, you can use jars and pickle food, you can smoke food, dry it over fire, tin-can it with a can sealer, you can make scandinavian lutefisk, and in this issue we try to think of new ways of preservation.

So with due respect - i do not agree with what you say in the first part, and I take that as your private opinion. I did my homework, realy. And as much as you don't like some people's opinion on how delicate food is, I as strongly dislike some folk's believes from other side of the spectrum on how unconditionaly long can food last. Nevertheless as I said I don't work on opinions, so none of that affected my work. Rant over.

As for the rest - great job. Great insight on food processing.
I'd like to know I you have any favourite food preserving process or processes that is/are not represented by the game as of now, and maybe worth implementing?

**Comment by tinukedaya** [comment]
@nexusmrsep 
first of all let me state that I'm happy with most of the changes that were introduced by you and I can easily survive the rest. You've done incredible job on your research and it's implementation to the game. 

But then I do agree with @Amariithynar on most points as well. 

My suggestion here would be to slightly change the long-lasting food. Most people argue that for example rice should be edible even long after it's shelf-life date. I agree with that and maybe there could be a way to get the best of both worlds? 

What if the long lasting food (the one you set to last for a year) would actually not rot straight away. It would go old, reducing it's (and any meal that is made using it) nutrients to half (?) AND get a small chance of rotting. So if you're unlucky your paste will get bad after a year, but other may last forever.
You are already adding the RNG to the initial spoil time which I think is awesome, so lets make this an additional step?

Also, some canned, pickled, otherwise preserved food should in my opinion get a VERY SMALL chance to get rotten as if not preserved, to simulate the fact that not always you manage to do this things right. I have seen few canned meat that when opened made you want to throw up. 

For the possible preservation methods, I believe the vehicles minifridge should get some love. As you want to put much more stuff into it now, maybe increase it's volume, or make a fullsize version that is using more power. What I would like to see is minifridge installed in floortrunk for people like me, who want to have stuff sorted... That way I can have my ingredients (pasta, rice etc.) in there and only meals ready to eat in the (mini)fridge.

Another addition i'd like to see is freezer to support the aforementioned idea of different flags ( frozen, cool, etc.). Food in freezer would last much longer than in fridge at the cost of higher power need AND making frozen food being unable to be used in recipes unless it thaws. That way, you might get highly perishable stuff that last as long as you have the power for freezer, but you have to plan ahead what you'd like to cook.

**Comment by sfsworms** [comment]
Couldn't we use the existing rot value to measure that? Lengthen the maximum rot time (to inedibleness), but gives maluses to nutrients and enjoyment when you have high rot value? You could add a "start_decay" modifier from 0 to 1 which would be how long the food remain perfectly edible (so a food with a decay mod of 0.8 would be perfectly fine up to 80% of its max rot, then linerarily loose nutrition and enjoyment until it's rotten).

**Comment by Amariithynar** [comment]
@sfsworms if that can be coded easily, then I definitely support that idea! Unbutchered carcasses would have a very low decay mod, since they decay very fast, while butchered meat (but not organs) would have a high decay mod (since they've been sectioned off from the entrails and drained), and preserved meats would be even higher, due to the preservatives used in their creation.

**Comment by TechyBen** [comment]
Would a half life model apply? Or simply an adjustment to the fresh -> normal -> old -> rot be needed?

**Comment by kevingranade** [comment]
> Here's the thing... Almost any spoil time information available is "food can be safely eaten with 100% confidence"

As you say, that number is all that seems to be available.  It's also exactly the number we want right now, because the game has zero chance for eating food making you sick or providing less than nominal nutrition, instead it simply disallows it.  
We are very likely to add support for eating food past that date at some point, with growing penalties and risks for doing so over time, but we are not going to change how we determine what that safe date is based on anecdotes.

> It would go old, reducing it's (and any meal that is made using it) nutrients to half (?) AND get a small chance of rotting.

If we add support for eating food past its "best by" date, it's not a simple matter of reducing nutrition and fun, it would also gain a growing chance over time to make the player sick, up to and including killing them.  It will still be the case that the only completely safe food to eat is food within the currently identified safe window.

**Comment by Amariithynar** [comment]
@kevingranade It's fine if that's the data that you choose to go off of rather than my personal knowledge and the knowledge of other cooks, butchers & chefs, as far as I'm concerned; so long as it's understood that that's not what actual *rot* times are for food, only the FDA's "best practices" to avoid potential issues in a commercial setting. Using them as "best by" dates isn't wrong at all, and would work with a refactoring of the system to allow edibility past the date with risks taken, anyways.

I think for altering the system to support edibility beyond best by dates, that the (old) tag should be repurposed for that window before it actually reaches rotten with a malus to enjoyability (except for meat-only meals possibly, as per how most people actually prefer an aged meat?), and altering food poisoning; changing it entirely from the binary system of "you're poisoned or you're not" to instead a system like radiation (with the counter being invisible), where the more food you consume that isn't fresh increments the counter with deleterious effects occurring at various break points (light food poisoning, moderate food poisoning, heavy food poisoning, etc.) would be for the best all around. That would also let the severity of the detriment vary depending on the foodstuff and good preparation methods could potentially wipe out the counter (like with dehydration and making salt jerky) from the raw ingredients. This would also give "Strong Stomach" more of a purpose, as you could resist food poisoning easier; a drect percentile reduction in food poisoning gain and/or faster reduction in gained food poisoning, for example (which would be representative of a particularly inhospitable gut biome for microbes). Combining that with lowering the nutrition/vitamin levels as prepared food ages would be pretty accurate as an abstraction of how it actually works in real life, while also not introducing too much makework for the player either.

I'd also recommend a knife action that allows you to trim off the outside from raw ingredients, potentially wasting the food ("it was rotten all the way through." "there just wasn't enough that wasn't rotten to use." "your knife slipped and cut into the organs; you can't recover any meat from this, now" and such) but also reducing its food poisoning level, modify the nutrition downwards (since there's less of the ingredient to go around; implimenting it as a penalty to nutrition stat that gets inherited by the prepared foods perhaps? Or just converting it to a generic meat/veg/fruit item called "pared down <blank>" that has reduced nutrition and uses more of them for the same recipes, if you'd rather retain the recipe framework with little additional work; this could also product as a byproduct "organic scraps" that you could toss into a shallow pit to compost!) using some formula that involves the Survivor's cooking level, with bonuses applicable for the food services industry professions perhaps, or just flatly resetting it (as the majority of the microbial effects on food are surface level; this is why bruising or a nick in most produce causes them to wither and rot substantially faster than undamaged produce, and why you can shave off the top inch or two of meat from a primal or subprimal cut that has been aging and the meat within is perfect for eating).

**Comment by nexusmrsep** [comment]
> It's fine if that's the data that you choose to go off of rather than my personal knowledge and the knowledge of other cooks, butchers & chefs, as far as I'm concerned;

Do you, as an expert in the field, know of any written and publicly avalaible and accessible written source of scientific info that reflects that knowledge?

> We are very likely to add support for eating food past that date at some point, with growing penalties and risks for doing so over time, but we are not going to change how we determine what that safe date is based on anecdotes. 
>If we add support for eating food past its "best by" date, it's not a simple matter of reducing nutrition and fun, it would also gain a growing chance over time to make the player sick, up to and including killing them. It will still be the case that the only completely safe food to eat is food within the currently identified safe window.

I'd gladly queue that idea for a future PR.
Also @kevingranade , I'd like to know your opinion on the freezer idea, perhaps working at large (x4 minifridge) power consumption. I can guess it is not in game for balancing reasons, but perhaps it can be introduced on certain conditions, like this extra electricity consumption. If not, that would be a solid info too.

> I'd also recommend a knife action that allows you to trim off the outside from raw ingredients, potentially wasting the food ("it was rotten all the way through." "there just wasn't enough that wasn't rotten to use." 

Low butchering skill = less meat, already in game. Also I'd prefere not to add to many empty micromanagement like removing skin from a carrot or somethin on the lines. If cooking recipe use cutting tools like knives you can easily guess such actions are accounted for. As for cutting spoiled parts to reveal rotten parts is not as universal to be added, and as some other ideas you propose here are oversimulating in my opinion. I will not quote all of them, but some are other ways to achieve what can be already achieve. I believe we don't want to promote competitive methods of the same principle or process of food preservation unless for a good reason. Batch recipes work magic.

> Also, some canned, pickled, otherwise preserved food should in my opinion get a VERY SMALL chance to get rotten as if not preserved, to simulate the fact that not always you manage to do this things right. I have seen few canned meat that when opened made you want to throw up.

Nice idea, and I believe possible to code.

> For the possible preservation methods, I believe the vehicles minifridge should get some love.

Nah, It serves a purpose: pressure on the player is good - it inspires to act. It gives limitations. Simple space limitation can force player for example to expand their deathmobile. It sure worked in my personal gameplay, and I was glad it did. If I had a huge fridge and a huge freezer at no lor low cost where would the fun be? Sometimes not doing something is for the better. (Not that I'm against adding a freezer with some extra limitations for balance.)

> Garlic cloves do not spoil. Since they're not seeds with a hard shell, just pieces of a garlic bulb which does spoil, they should spoil as well.

True, I'm queueing it.

> Many vegetables cannot be preserved in form other than generic vegetable matter - which is unsuitable for many recipes unlike normally preserved form of particular vegetables would be. Checking generic-making recipes like "pickled vegetable", "canned veggy" and "vacuum-packed veggy slice" may be useful here. While some recipes using particular vegetables also use the generic preserved kind, some vegs that can be "generalized" may or may not be usable directly in some of those recipes when they should/shouldn't (depending on the recipe).

Any idea to mitigate it, other then spamming lot of additional recipes?

> Lastly, in reference to great @Amariithynar post, I'd like to suggest making meat spoil slightly slower when it's separated from guts and other such bits than when it is a part of the corpse, providing benefits to butchering shortly after hunting an animal, rather than it making no difference to shooting up a forest for a day and then processing the corpses on one's way back as it's currently most convenient in game.

Corpses rot too, so how about: 
- [option 1] passing it's relative rot (in %) to meat's relative rot (in %), or 
- [option 2] adding a chance based on relative rot of the corpse (in %) that buthered meat will be already rotten?

**Comment by RadHazard** [comment]
>>Also, some canned, pickled, otherwise preserved food should in my opinion get a VERY SMALL chance to get rotten as if not preserved, to simulate the fact that not always you manage to do this things right. I have seen few canned meat that when opened made you want to throw up.

>Nice idea, and I believe possible to code.

If this gets implemented, the chance should definitely depend on cooking skill

>>Many vegetables cannot be preserved in form other than generic vegetable matter - which is unsuitable for many recipes unlike normally preserved form of particular vegetables would be. Checking generic-making recipes like "pickled vegetable", "canned veggy" and "vacuum-packed veggy slice" may be useful here. While some recipes using particular vegetables also use the generic preserved kind, some vegs that can be "generalized" may or may not be usable directly in some of those recipes when they should/shouldn't (depending on the recipe).

>Any idea to mitigate it, other then spamming lot of additional recipes?

My suggestion would be to change the various preservation methods to use a different system rather than the crafting menu.  The simplest way would be to let players craft generic "pickling jars", "vaccum bags", etc. and then have the player `a`ctivate them to preserve anything they want.  Flags could be used to control which kinds of food can be preserved by which method to stop nonsensical combinations.  This would actually play well with the idea of having occasional "canning failures", since I don't think it's possible to bake that kind of thing into the crafting system directly.

**Comment by kevingranade** [comment]
> Also @kevingranade , I'd like to know your opinion on the freezer idea, perhaps working at large (x4 minifridge) power consumption.

It's not missing due to balance concerns, just due to lack of implementation.  I haven't put much thought into it, but I'm not really sure it even has that much more power draw or needs it.  We don't freeze all our food IRL because it has many practical drawbacks, we just need to capture enough of those to make it work as expected.

Random additional comment about refrigerators and freezers, home appliances tend to be grossly inefficient compared to travel versions, and incredibly heavy to boot.

RE: trimming rotten pieces off of food, I definitely dont want that to be a distinct action.  To the extent that it makes sense to do that, it should be possible to add it to other actions like crafting or consumption.  I dont know how generally applicable a thing that is to do to rotten foods, and I'm not sure it's going to be worth it to annotate a lot of items to make it happen.  I'm not against the concept, but it seems like it might be a lot of work.

> Many vegetables cannot be preserved in form other than generic vegetable matter

It *might* be reasonable to embed the base food type in the results, so you get, "pickled vegetable (carrot)", which might be usable in some carrot-specific recipes.

> The simplest way would be to let players craft generic "pickling jars", "vaccum bags", etc. and then have the player activate them to preserve anything they want.

That's not how that works, you need to cook/pasteurize/ferment the food first, which has little to do with the container.  It should be possible (maybe tricky though) to make recipes do the right thing here.

> I'd like to suggest making meat spoil slightly slower when it's separated from guts and other such bits than when it is a part of the corpse,

I could have sworn I opened an issue about this.  Field dressing, discards less valuable parts, makes the corpse more compact for carrying, and delays spoilage.  My understanding is it's fairly rare to butcher kills immediately when hunting so something like this is very typical.

**Comment by RadHazard** [comment]
>That's not how that works, you need to cook/pasteurize/ferment the food first, which has little to do with the container. It should be possible (maybe tricky though) to make recipes do the right thing here.

Right.  My intent was the "use" action would require the appropriate tools and time to represent those steps.  Come to think of it, the use action could just be added to the raw container items without an intermediate crafting step in most or all cases.

If you can make them work, recipes would be fine.  My only concern is that if we want to avoid cluttering up the menu with a bunch of recipes for every single item, we'd need to implement a way for recipes to "carry through" one or more ingredients.  I haven't dug through the recipe code so I'm not sure how hard that would be to implement vs. a "preserve food" use action.

**Comment by TechyBen** [comment]
> I could have sworn I opened an issue about this. Field dressing, discards less valuable parts, makes the corpse more compact for carrying, and delays spoilage. My understanding is it's fairly rare to butcher kills immediately when hunting so something like this is very typical.

@kevingranade Yes, Ray Mears did this with a stone knife in like 20-30 mins. Made some nice "deer chops" and "deer wings" to carry with him (no modern hygiene needed). (5 mins video clip : https://youtu.be/Th6EOlLK0DA?t=34m59s )

This would need training/experience. But after that, it would be a time and weather (rain/heat?) based limiting factor.

As I've mentioned before, signposting to players seems fine. If these foods are given lesser lifespans, then "it's dirty" or "it smells bad" (as it currently has) style flavour/lore text matches better than "it's rotten". Though, saying that, even quick rot food systems, may force players into managing eating rotten foods in emergencies (as in real life!). Would be an interesting mechanic if food supply balance in game went in that direction (due to zombie risks, or just supply/demand balance).

**Comment by FulcrumA** [comment]
@Amariithynar 
>You'd make the good food when you're safe at base, and only enough to feed yourself in the short term. The rest would be aged, dried, converted to jerky, etc. for longer-term storage.
I think that with current, new spoilage system, we're getting there naturally, even without new food poisoning - you can only make so many sandwiches or other food before it'll get spoiled before you'll eat it all. 
Not that non-binary food poisoning is a bad idea - I think it'd be nice to have it as some value increased due to eating some raw food (but only at random, it should still be a viable choice to eat raw, just with some risk the way it is now) and some spoiling (with "old" providing far less of "food poisoning points" than "rotten").

**Comment by kevingranade** [comment]
> considering we can put tank cannons on our vehicles, I doubt the weight of a house fridge would mean much to the average death mobile

If you can manage to get a tank type vehicle working, yes a refrigerator isn't going to be much of a challenge, but expect this to get harder to accomplish over time.

> and laser cannons with quantum solar panels I don't think the power draw of a fridge would hamper much.

Same situation here, power production is currently broken, expect it to be fixed in the future.

**Comment by nexusmrsep** [comment]
I also don't understand how food poisoning can be non-binary. You are either sick or not. Either your stomach and guts digest the food, and fight the bacteria or not. I see it more like a statistical chance based on strain that you force on your immune system. Let say your immune system is at nominal 100 value. If you eat large amount of maggot infested, mold ridden, rotten to the bone meat, it would count as lets say -150 value, so it blasts thru your system like a bacteriological bullet of a highest caliber, giving you no chance but to make world record attempts at projectile vomiting. But if you eat a bite of a slightly off venison that you held on for a day too long, that might be considered at lets say -25. 100 - 25 = 75 % to stay healthy and 25 % to spend the rest of the night in the outhouse. Also, your immune system got worse from chomping down the stinking venison, so if you are brave enough for another bit its a 75 - 25 = 50%. And now it fifty fifty, and you might not like those chances any more. 

So at the end you get 0-1 system for the illness, and non-binary for the chance of it happening.
Also lets say there is a threshold of lets say minus 100 (values from the moon, don't consider them anything meaningful), so if you eat (100-200=-100) or (25-125=-100) or (50-75-75=-100), etc., you get lethal version of poisoning. Also if you already suffer food poisoning and still eat rotten food, it might end with the same result.

I'm not 100% sure if current hidden health value would be a good measurement for above, but it sounds as a good lead.

**Comment by Tharn** [comment]
One way I could see it being non-binary: Different foods carry different bacteria. They also rot differently. Meat and eggs for example are known to easily become dangerous when they go off; even when fresh, you may need to cook them properly to not risk an illness. Fruits and veggies seldom are that dangerous even after they spoil. The risk here lies more in contamination with animal feces as fertilizer or during processing. Home-grown plants in Cataclysm should have a very low chance of causing food poisoning. Higher if fertilized or if the ground is contaminated by blood, bile, etc.

**Comment by TechyBen** [comment]
Yeah, don't quote me on hard figures here, but some food poisoning can kill you, others just make you feel queasy, others make you go the loo more often.

If it was linked to food portion size, so eating a tiny bit of old food having a small effect vs a whole rotten lasagne (though some chars just would not be able to "force" the eating, till at least "starving" status).

**Comment by Amariithynar** [comment]
>That's not how food poisoning works, it's generally not a poison, it's a chance of being infected. Theres no safe amount of risky food you can eat, every time you eat risky food there is a chance of getting sick.

Actually, it is, for the most part. All food has bacteria and other microbes on it. The reason why food is considered "spoiled" is because it reaches a critical number of bacteria, which has a higher chance to overwhelm your system, but it's not guarenteed. Every time you consume something, your body's doing its duty, fighting off the bacteria. That's why HIV/AIDS is so deadly; it severely weakens the immune system, and you need to be extraordinarily careful of what you eat and how you eat it, so you don't introduce more to your weakened system than it can handle. Same with when you're already sick, your body comes to tolerate less of a range of foods and food qualities. These concepts are already (very) abstractly given in game via the "strong stomach" and "weak stomach" traits as well as "poison resistant", in that you're less likely to vomit (instead of purging the contaminated material, just able to fight it off) and you have a higher tolerance to poisoning, as well.

What IS binary is not whether the bacteria are there, but whether they've been successfully fought off. If they haven't, that's mild food poisoning. Your threshold is then weaker to the next bacteria, as they're still trying to fight off the first wave. This makes it easier to get a longer lasting case, or for another strain to set up inside of you as well. It's also not a one-and-done sort of check, but the bacteria constantly multiplying and either they release toxins to inhibit your immune system like campylobacter, where they can survive for weeks at a time before you suffer from the deleterious effects of food poisoning, as they gain critical mass inside your body rather than before ingestion, or things like salmonella, that just directly release toxins into our systems that weaken us by giving us the runs (the body reactively trying to cleanse the infection by flooding it out) and vomiting, losing a lot of nutrients and more importantly fluids which diverts resources from going to fighting the infection, so it's easier for them to multiply. E. coli strains release a verocytotoxin which damages blood cells and can cause irreparable damage to your kidneys, which can itself lead very quickly to death; it's naturally present in every animal's gut flora, too (which is why when making sausages you need to thoroughly clean them, invert them (exposing the inner surface they live on to the elements), scrape off the mucal layers (where most of the bacteria reside) and then salt-cure them to ensure no bacteria remain. Botulism's toxin seeks out nerve cells to infiltrate along a specific receptor path and damage the protein structure needed for the cell to communicate with others, which causes paralysis; if this happens to the heart, or the lungs, or the brain... well, you're dead. That's why botulism is so deadly, by the way. Not because it's hard to fight off, but because the toxins released are so potent that it only takes a little to cause a great effect. It's the toxin that kills you, not an overabundance of botulism. That's also why pressure-cooking and canning hve to be done right, because if you don't, then you don't destroy the spores and they produce more toxin. The toxin itself is rendered inactive by boiling for ten minutes.

Long and short of it is this: Modelling all that on different levels depending on what you eat is way too complex. That's why I gave the example of an abstraction to a rad-like value that increases as you consume foods and decays over time (representing your body successfully fighting off the infection of bacteria), but adding an influx of bacteria also means that your body is harder pressed to fight them, and the first symptoms can show up. Some bacterial colonies are so potent that they only need a little to infect you, so they'd have proportionately high 'food poisoning' values. Others would be low grade, but would last for a long time, passively reducing your natural food poisoning decay rate and increasing by small amounts over that long time, and so on and so forth.

**Comment by Tharn** [comment]
Another small idea. Assume you have a single stack of food that lasts 3 years. You put a third of it in the fridge, then an hour later you put another third in the fridge, then another hour later another third. You now have 3 stacks.

It would be nice if long-lasting foods made spoilage checks more 'lazily' like once every day, so that player processing or very slight handling differences did not result in so many stacks.

**Comment by Kelenius** [comment]
>Assume you have a single stack of food that lasts 3 years. You put a third of it in the fridge, then an hour later you put another third in the fridge, then another hour later another third. You now have 3 stacks.

This is the main reason why I think that long-lasting food should be just immortal.

**Comment by Amariithynar** [comment]
I like the idea of a protein/fats/carbs system, but it'd have to be handled properly. High carb food is great for energy stores, which would result in pretty quick saturation from high carb sources like potatoes. However, carbs also get digested a lot faster than proteins or fats do, which provide longer-lasting but slower energy gains. This should tie into the stamina system directly, as well, which honestly needs to be made a larger range, so that iit takes a longer to recover to peak stamina, but you can increase your recovery rate in the short term by ingesting more carbs, and over the long-term by ingesting more protein; fats are more of a mid-term, as they are a carbohydrate, but they're a complex one; a triglyceride that needs to go through lipid metabolism.

Furthermore, a high-carb diet that doesn't expend most of the excess calories consumed would lead to obesity and general health issues, just as a high-protein diet would lead to constipation, kidney stones, and osteoporosis. We could simulate this imbalance through temporary use of traits that already exist in the game, with the "Languorous" trait http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Languorous for high-carb diets, and high-protein diets could get "Meat Intolerance" http://cddawiki.chezzo.com/cdda_wiki/index.php?title=Meat_Intolerance as their gut acts up and they have issues with digestion. 

>a diet far more balanced and healthy than how many people live on average, in game still leads to considerable, even crippling deficiencies quite quickly.

This really does need to be taken into consideration, though, first and foremost, if the current vitamin system remains. Honestly, though? I'd rather see it abstracted out to the common food groups- grains, meats/legumes, dairy, vegetables (legumes also count here), and fruits. This would also further encourage static bases, and you couldn't just pop vitamins all the time to have a balanced diet. This would give further strength to "deluxe" food items, and encourage people to make the higher-tiered food, as they contain a larger variety of ingredients in any given portion. 55g of dandelion greens, for example, contain all the vitamin A you need in a day, and 32% vitamin C, as well as 10% and 9% respectively of calcium and iron, so it's not like you NEED fruits over vegetables, even to deal with scurvy. http://nutritiondata.self.com/facts/vegetables-and-vegetable-products/2441/2 It would also deal with the issue where all baseline-edible 'meat' is considered the same item, without the need to add a whole bunch of variants for accuracy, and mean that you can't just subsist on nothing but bread from grains; you have to diversify your meal plan.

**Comment by FulcrumA** [comment]
Kinda sorta chipping in with a new idea regarding preservation. I've read around and supposedly aluminum cans can be used food preservation. Perhaps they could be used as an alternative to tin ones (with addition of 2, 3 pieces of scrap iron more since they'd probably use some tinkering to fit the role)? during the game aluminum cans keep piling up without having too much use when they easily could.

Alternatively, ability to slightly tinker with them and some soldering iron/welder could allow turning cans into metal canisters - which then could be used both for food preservation and, added bonus, other improvised things using those small metal canisters.

**Comment by FulcrumA** [comment]
Looking forward to it, @nexusmrsep - I assume you mean to do it as a new feature? 
I had an idea of doing something like that as a json based crafting recipe (since json edits are something I am actually still willing to work with as compared to my questionable programming experience most of which I lost anyway over the years) but I couldn't think of it in a way that wouldn't be clunky and mess with butchering ability of tools.

**Comment by nexusmrsep** [comment]
Yes, new feature with nothing to do with recipes

**Comment by KorGgenT** [comment]
number 13 is addressed by #28521

**Comment by KorGgenT** [comment]
also number 8 is partially enabled by #27252

**Comment by Tharn** [comment]
It looks like nuts from nut trees do not expire currently. This is incorrect. They last a while, but they expire because the fat goes rancid. Here's a great resource that you can use for various nuts: http://www.eatbydate.com/proteins/nuts/how-long-do-nuts-last-shelf-life-expiration-date/

Irradiated fruit also currently does not expire, which I assume is just because no one has looked at it yet. I would blanket set irradiated fruit to last 4 weeks longer than normal fruit. This seems to be a reasonable (probably too reasonable) median, but in the grand scheme of CDDA it's still not that long.

**Comment by nexusmrsep** [comment]
This tracker issue is now obsolete, as majority of the ideas were done and merged, and even built upon. I may return to carbs/protein/fat idea later. Closing.
